kernel/thread.c

```diff
@@ -523,8 +523,21 @@ void _k_thread_group_leave(u32_t groups, struct k_thread *thread)
 }
 
 #ifdef CONFIG_USERSPACE
+FUNC_NORETURN void k_thread_user_mode_enter(_thread_entry_t entry,
+					    void *p1, void *p2, void *p3)
+{
+	_current->base.user_options |= K_USER;
+	_thread_essential_clear();
+	_arch_user_mode_enter(entry, p1, p2, p3);
+}
+
 int _is_thread_user(void)
 {
 	return _current->base.user_options & K_USER;
 }
-#endif
+
+int _is_user_context(void)
```

Comment:
Same comment as on the wrapper: seems like this really wants to be an inline so that code that needs to use it (which is really likely to be on a hot path) can get the one-instruction version instead of the double-wrapped-function-call.

Reply:
agreed